### PR TITLE
React & Redux group project - Space Travelers' 

### PR DESCRIPTION
**Pull request by omarsalem7**
## This pull request includes all the functionality needed for the project to be approved.

We have used the SpaceX API to get a list of missions, rockets, and dragons. We have used redux to save the data in the state.

## Missions Page
- listed all missions.
- in addition to the mission name and description we added a chip for status and a button for joining and leaving the mission.
- we used redux action to toggle the reserved state of the selected mission.

## Rockets Page
- listed all rockets.
- in addition to the rocket image, name, and description we added a chip for status and a button for reserved and cancel.
- we used redux action to toggle the reserved state of the selected rocket.

## Dragons Page
- listed all dragons.
- in addition to the dragon image, name, and description we added a chip for status and a button for reserved and cancel.
- we used redux action to toggle the reserved state of the selected rocket.

## My Profile Page
- show "no missions joined or no rockets reserved" if nothing is revered.
- filter the list of missions to show the joined missions.
- filter the list of rockets to show the reserved rockets.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,10 +1,33 @@
...
+import { useDispatch } from 'react-redux';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
+import { fetchRocketsDispatch } from './redux/rockets/rockets';
+import { fetchDragonsDispatch } from './redux/dragons/dragons';
+import Header from './components/header/Header';
+import Profile from './components/profile/Profile';
+import Rockets from './components/rockets/Rockets';
+import Missions from './components/missions/Missions';
+import Dragons from './components/dragons/Dragons';
 import './App.css';
```
Comment (rloterh) [inline src/App.js:12]: - [ ] Test cases are missing. Kindly ensure to write unit tests with React Testing Library and Jest (for unit tests)

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Josphat & Omar, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Good Points 👍  
- Descriptive Pull Request
- Correct gitflow used
- State management with Redux properly implemented
- App is running as intended
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 4 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 1 above)
Comment (omarsalem7) [reply src/App.js:12]: Hello @rloterh 
The testing is not required for the project to approve

Comment (omarsalem7) [reply src/App.js:12]: Hello @rloterh 
I added the test cases, you can check it now
Thanks,

**Review by Meltrust — APPROVED** [review]
Hi @omarsalem7,



## Wow, you did it 🎉
image
Brilliant
## Thank you for implementing the changes requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**